packages/cache/src/cache.ts

```diff
@@ -576,7 +589,7 @@ async function saveCacheV2(
     } else if (typedError.name === ReserveCacheError.name) {
       core.info(`Failed to save: ${typedError.message}`)
     } else {
-      core.warning(`Failed to save: ${typedError.message}`)
+      core.error(`Failed to save: ${typedError.message}`)
```

Comment:
Add a specific condition for server errors (5xx) that will emit an error log and keep all the other errors as warnings. Fix all the unit tests to match this behaviour.